loki/batch/scheduler.py

```diff
@@ -495,7 +498,7 @@ def _get_definition_items(_item, sgraph_items):
                     _item.scope_ir, role=_item.role, mode=_item.mode,
                     item=_item, targets=_item.targets, items=_get_definition_items(_item, sgraph_items),
                     successors=graph.successors(_item, item_filter=item_filter),
-                    depths=graph.depths
+                    depths=graph.depths, build_args=self.build_args
```

Comment:
[no action] I would argue that we could also attach the `build_args` to the `Item`, so that we continue to use the `Item` as the universal meta-data object that gets handed from `Scheduler` to `Transformation`? Admittedly, this can be said for almost any of the arguments here - so this might also be part of a larger "scheduler-item" refactoring and need not go into this PR.

Any strong opinions on this, @reuterbal ?

Reply:
Yes, good idea.

The only thing where we need to be a little careful is that the `definitions` part is updated dynamically when calling `_parse_items`. When re-parsing due to topology changes that introduce a new dependency that was previously not part of the graph, these definitions would then be missing for existing items. Hence, this is probably out of scope for this PR.

For the other arguments it depends. `role` and `targets` should be used from the Item and have only been kept for backwards compatibility. We should indeed remove them here. The other two are properties at a graph-level, not an item level, so there's an argument to keep them separate.

Reply:
Ok, agreed. I'll look into this in a separate PR. 👍